bin/semver.js

```diff
@@ -88,7 +89,7 @@ const main = () => {
     }
   }
 
-  options = { loose: loose, includePrerelease: includePrerelease, rtl: rtl }
+  options = parseOptions({ loose, includePrerelease, rtl })
```

Comment:
why create a new object here at all if the goal is to avoid new object creation?

Reply:
This method executes only once, I don't think is a problem to keep it this way, also, I put parseOptions here to reduce the number of calls for `parseOptions` using the original object (which adds more comparisons).

Reply:
I agree with @ljharb. If we are doing micro-optimizations, creating an extra object has a cost too. It would be faster to pass it as different arguments.

Reply:
I wonder if the following approach would be faster:

```
parseAndMutateOptions(options, loose, includePrerelease, rtl)
```

And instead of returning an object, and later copying it to a different variable, the function can directly mutate the `options` variable avoiding the extra copy.

Reply:
With the new behavior of returning the object in parseOptions, this function doesn't allocate extra memory and just returns the same object.

> parseAndMutateOptions(options, loose, includePrerelease, rtl)

From what I remember, this could be faster but also can suffer from deopt if some of those arguments are passed with different types, like `undefined`, `true`, etc...

> And instead of returning an object, and later copying it to a different variable, the function can directly mutate the options variable avoiding the extra copy.

Usually, this is a good approach but I don't think that is a good behavior to introduce without marking it as breaking change, but I will keep it in mind in case we are forced to return booleans.

Reply:
Mutating the options seems like a bad idea; it seems quite possible that those options to have come from something that the caller thought would remain unmodified.

Reply:
In light of https://github.com/npm/node-semver/pull/535#issuecomment-1499331981 I think we can leave bin/semver.js alone. It's fine as-is and isn't exactly a hot path.